### DISCUSSION TITLE
How to deploy large CloudFormation templates?

### DISCUSSION THREAD

**Discussion by mvs5465** [aws/aws-cdk#23515] [Q&A]
What is the expected way to deploy large stacks from CDK?

```
Resource handler returned message: "Model validation failed (#/TemplateBody: expected maxLength: 51200, actual: 56977)"
```

I am deploying a [Conformance Pack](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_config.CfnConformancePack.html) with a large number of rules and the template exceeds the CloudFormation request limit. 

I've already separated the conformance pack into a NestedStack but I don't think I can divide it up more than that. The prescribed solution to this is to upload the template to an S3 bucket and use that as the source of the template body.

I'm sure this is a duplicate but I searched discussions and issue reports and couldn't find anything.

Thanks!

**Comment by khushail** [comment]
Hi @mvs5465 , the possible way to deploy a large cloudformation template are suggested [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html)-
1. split the template into smaller templates based on their logical grouping. 
2. upload the template file to S3 Bucket and reference the template URL in  AWS CDK Code.

Hope that is helpful.
